packages/frontend/src/components/Modal/AccountDetailModal.tsx

```diff
@@ -0,0 +1,113 @@
+import Jazzicon from '@metamask/jazzicon'
+import { shortenAddress, useEthers } from '@usedapp/core'
+import copyToClipboard from 'copy-to-clipboard'
+import { useEffect, useRef } from 'react'
+import { CopyIcon, RedirectIcon } from 'src/components//Icons'
+import { Button } from 'src/components/Buttons/Button'
+import { ContentRow, Modal } from 'src/components/Modal/Modal'
+import { useChainId } from 'src/hooks/chainId/useChainId'
+import { Colors } from 'src/styles/colors'
+import { getExplorerAddressLink } from 'src/utils/getExplorerLink'
+import styled, { css } from 'styled-components'
+
+export interface ModalProps {
+  isShown: boolean | undefined
+  onRequestClose: () => void
+  wallet?: string
+}
+
+export const AccountDetailModal = ({ isShown, onRequestClose, wallet = 'Metamask' }: ModalProps) => {
+  const { account, deactivate } = useEthers()
+  const accountIconRef = useRef<any>(null)
+  const chainId = useChainId()
+
+  useEffect(() => {
+    if (account && accountIconRef.current) {
+      accountIconRef.current.innerHTML = ''
+      accountIconRef.current.appendChild(Jazzicon(40, parseInt(account.slice(2, 10), 16)))
+    }
+  }, [account])
+
+  return (
+    <Modal isShown={isShown} onRequestClose={onRequestClose} title="Your account">
+      <ContentWrapper>
+        <ContentRow>
+          <ConnectedWallet style={{ margin: 0 }}>Connected with {wallet}</ConnectedWallet>
+        </ContentRow>
+        <ContentRow>
+          <AccountIcon ref={accountIconRef} />
+          <AccountAddress>{shortenAddress(account || '')}</AccountAddress>
+        </ContentRow>
+        {account && (
+          <ContentRow>
+            <AccountLink
+              href={account && getExplorerAddressLink(chainId, account)}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <RedirectIcon size={24} color={Colors.Blue} />
+              View in block explorer
+            </AccountLink>
+
+            <CopyAccount onClick={() => copyToClipboard(account)}>
+              <CopyIcon size={24} color={Colors.Blue} />
+              Copy address
+            </CopyAccount>
```

Comment:
Right now, clicking "Copy address" closes the modal. After fixing the previous comment it won't, so I recommend adding a tooltip "Copied!" after the user clicks "Copy address"